src/components/ProposalCard/ExpandedActionsList/ExpandedActionsList.tsx

```diff
@@ -10,29 +10,31 @@ import {
 import { ExpandedActionsListProps } from './types';
 
 const ExpandedActionsList: React.FC<ExpandedActionsListProps> = ({
-  actions,
+  option,
 }) => {
   return (
     <ActionsTooltipWrapper aria-label={'expanded actions list tooltip'}>
-      {actions.map((action, index) => {
-        const InfoLine = getInfoLineView(action?.decodedCall?.callType);
+      {option?.actions.map((action, index) => {
+        const decodedAction = option?.decodedActions[index];
+
+        let InfoLine = getInfoLineView(decodedAction?.decodedCall?.callType);
 
         return (
-          <Fragment key={action.id}>
+          <Fragment key={decodedAction?.id || index}>
             <ActionWrapper>
               <ActionNumber>{index + 1}</ActionNumber>
               {!!InfoLine ? (
                 <InfoLine
-                  decodedCall={action?.decodedCall}
-                  approveSpendTokens={action?.approval}
+                  decodedCall={decodedAction?.decodedCall}
+                  approveSpendTokens={decodedAction.approval}

```

Comment:
I see you refactor this from `action?.approval`. Can `decodedAction` be null at some point? If so, please add `.?` as it was before.